src/rpc/util.cpp

```diff
@@ -860,53 +880,75 @@ void RPCResult::ToSections(Sections& sections, const OuterType outer_type, const
     NONFATAL_UNREACHABLE();
 }
 
-bool RPCResult::MatchesType(const UniValue& result) const
+static const std::optional<UniValue::VType> ExpectedType(RPCResult::Type type)
 {
-    if (m_skip_type_check) {
-        return true;
+    switch (type) {
+    case RPCResult::Type::ELISION:
```

Comment:
```suggestion
    using Type = RPCResult::Type;
    switch (type) {
    case Type::ELISION:
```

nit: not sure if it compiles, though